### PR TITLE
proofread readme documentation and correct english terms

### PR DESCRIPTION
**Pull request by mattma**
校正英语语法，因为我对这个 Project 不是很了解，可能有的地方会有出入，但希望能够有帮助。请到［这里］（https://github.com/mattma/egg-loader）看看 rendered markdown file.

---
This change is [Reviewable](https://reviewable.io/reviews/eggjs/egg-loader/7)

### CODE REVIEW

**Comment by mattma** [comment]
@fengmk2 我刚开始看 egg project, it is pretty cool. I am going to review more readme docs in the following weeks. For this repo, I feel like it is a loader like Webpack, it can load up different environment configurations in the runtime. am i right?

**Comment by popomore** [comment]
Good, welcome to help us to translate.

File: README.md [hunk 1]
```diff
@@ -132,11 +130,11 @@ Load agent.js
 
 Load app/service
 
-### Low Level API
+### Low Level APIs
 
 #### getServerEnv()
 
-Get serverEnv for application, available serverEnv
+Retrieve application environment variable values via an object `serverEnv`. You can access directly after instantiation by calling `this.serverEnv`.
```
Comment (popomore) [inline README.md]: `serverEnv` is not an object.

**Comment by popomore** [comment]
Commit message should follow https://github.com/eggjs/egg/blob/master/CONTRIBUTING.md#commit-提交规范

**Comment by mattma** [comment]
@popomore 我们以前在 Slack 上聊过，想想都是一年多前的事了。

我已经更改了 `serverEnv` 翻译，加上了 `doc:` prefix before commit message, then squashed commits. 

I wish that I can contribute to the project in the future once I understand the purpose, use case of egg. By reading the docs, I feel like it is a server side framework like Express, Koa, but take some best practices from front end like webpack, cli tooling, etc, bring some goodness to Node.js development.

**Comment by atian25** [comment]
egg is a web framework base on koa, just like connect vs express

**Comment by mattma** [comment]
@atian25 That is awesome. I do think Server side framework need a facelift at some point. I am glad that you guys pioneer and lead us to the future! 

Have you ever considered to use typescript instead of vanilla js? Typescript becomes so popular recently here in US.

**Comment by popomore** [comment]
@mattma 我们在应用有使用 ts，但框架使用可能还不是时候（也还没有使用 koa2），我们会考虑定义 tsd。

**Comment by mattma** [comment]
The main benefit of taking advantage of TS is to ease the development pain. Ex: GoLang is a compiled language. All source code will be compiled through compiler so that you know your code won't run in your development time before it ships to production. TS is trying to guard the best practice for loose typed JS language and brings in the smart text editor feature to benefit developer while you writes your code in confidence. 

I have tried with TS with KOA2, and it works pretty good so far. The rising of Angular2, which built on top of TS, is leading the JS revolution wave. More and more developers have seen, used, enjoyed TS in their daily dev workflow. 

BTW, how is my PR? ready to merge? @popomore

**Comment by mattma** [comment]
@popomore For TS, typescript 2.0 is going to be released very soon. and They have updated their typings strategy. Read [here](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/)  Include a `.d.ts` is pretty easy. 

`tsd` and `typings`, that you mentioned above, are old good days. `tsd` project has been deprecated for 6 months already. When you use `typescript` with `declaration: true` flag should auto generate `.d.ts` files. 

Then you install the module through NPM, then definition file should be picked up by typescipt automatically

**Comment by popomore** [comment]
It seems litter misunderstanding. I mentioned the word `tsd` that's mean I will define a d.ts file for egg. 

I have used Typescript 2.0 in my work, and will use npm instead of the old typing modules, npm is awesome. What a large cost for Microsoft publishing for many `@type` scope package.

**Comment by mattma** [comment]
Cool. Good to know. 

Microsoft only drives the development effort of typescript. @type is run and maintained by community, majority of type definition should be merged over from DefinitedTypes project. 

BTW, if I understand correctly, egg project is an open sourced project by Ali, right? or a fun project by the top 10 JS developers in China, Gang of four? @popomore

**Comment by atian25** [comment]
an opensource project of ali, it is used at almost all the node project of ali.

**Comment by mattma** [comment]
has egg been stress test in production environment? Is egg ready to be built on large production Node.js server and runs on many machines?

**Comment by fengmk2** [comment]
@mattma Invite sent!

Welcome on board!

**Comment by fengmk2** [comment]
> has egg been stress test in production environment? Is egg ready to be built on large production Node.js server and runs on many machines?

@mattma We will share the detail datas on ningjs.

**Comment by fengmk2** [comment]
> 英文不太会写，不过 @mattma 可以双语交流，我就说中文。

我们在2013年就开始使用 egg 的前身，在2014年D2上分享过 chair http://yq.aliyun.com/articles/2921 ，然后 2015年我们成立了 ali node team 虚拟小组，一起共建内部版本的 egg。

核心：egg-core （现在叫 egg-loader，很快就会改名 https://github.com/eggjs/egg-loader/pull/6） 是一个框架的框架，它可以让有能力的开发者基于 egg-core 搭建符合自己团队需求的 web 框架。

也就是这个 repo 是整个 eggjs 最核心的部分。

- 关于 ts，我们未来肯定会支持，但现在还是先采取保守模式，正如我们现在只是基于 koa1 来做。因为我们不想任何 build 过程，我们希望还是 node server.js 这样就能启动一个服务。
- @mattma 如果你所在的团队或者公司，也希望做属于自己的符合自己业务需求的框架，基于 egg 也是很快能够封装起来。
- 关于性能和是否已经在生产环境使用。我在2011年加入阿里，就一直在做 node web 相关应用产品和技术，阿里应该是业界最早的一批使用 node 来做线上大流量应用的公司，只是因为相对于阿里的主流 Java 技术，node 的流量实在太小，就没怎么过多宣传。egg 这套技术从2013年起到现在有2年多的线上稳定运行经历，所以生产环境用是可靠的。性能数据我们在开源之前也会通过购买一台阿里云的 ECS 标准服务器来做 benchmarks，所以到时候能够看到真实一点的数据。

**Comment by mattma** [comment]
This is awesome! First of all, I am totally on board with this idea and I think that I could contribute to the project with my extensive deep knowledge on Node.js and JavaScript. 

I speak fluent chinese, A.K.A, Beijingish. But i am sucks on typing Chinese with PinYin. I do prefer to write in English, just for the sake of saving times. 

I will go through each repos soon, and share my thoughts in the following days. @fengmk2 It is honor to be part of ali open source project and work with such a strong Node.js team on earth. Thanks for the opportunity and cannot wait to see each other in person near future!

**Pull request merged**